cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -515,6 +534,30 @@ impl<I: VCodeInst> MachBuffer<I> {
         // Post-invariant: as for `put1()`.
     }
 
+    /// Begin a region of patchable code. There are some requirements for the
+    /// code that is emitted:
+    ///
+    /// 1. It must not introduce any instructions that could be chomped
+    ///    (branches are an example of this)
```

Comment:
I think branches are the only case actually; and at the `MachBuffer` layer all that matters are branches that we tell the `MachBuffer` about, via `add_cond_branch` and `add_uncond_branch`. So I think we can make this more concrete by saying "It must not introduce any MachBuffer-visible branches (i.e., you must not call `add_cond_branch` or `add_uncond_branch` between `start_patchable` and `end_patchable`)."

Reply:
I've updated the language on this comment, what do you think?